### PR TITLE
PPTP-1211 - simplify SubscriptionStatusResponse

### PR DESCRIPTION
**Pull request by SteveSugden**
### Description of Work carried through

Create a simplified SubscriptionStatusResponse that is aligned with the needs of the FE service

#### Check list 
 - [x] `./precheck` was executed (Integration/Component/Unit tests)
 - [x] `sbt scalafmt test:scalafmt` was executed
 - [ ] Required Environment Config has been amended/added
 - [ ] Links to dependencies have been included (BE/FE work)
 - [x] User Acceptance Tests (UAT) were run locally and have passed.

### CODE REVIEW

File: app/uk/gov/hmrc/plasticpackagingtaxregistration/connectors/SubscriptionsConnector.scala [hunk 1]
```diff
@@ -43,16 +43,23 @@ class SubscriptionsConnector @Inject() (
...
     val correlationIdHeader = correlationId -> UUID.randomUUID().toString
 
-    httpClient.GET[SubscriptionStatusResponse](appConfig.subscriptionStatusUrl(safeId),
-                                               headers = headers :+ correlationIdHeader
-    )
+    httpClient.GET[ETMPSubscriptionStatusResponse](appConfig.subscriptionStatusUrl(safeId),
+                                                   headers = headers :+ correlationIdHeader
+    ).map(etmpResponse => SubscriptionStatusResponse.fromETMPResponse(etmpResponse))
+      .recover {
+        case e =>
```
Comment (SteveSugden) [inline app/uk/gov/hmrc/plasticpackagingtaxregistration/connectors/SubscriptionsConnector.scala:59]: Q. Is this recovery a step too far?
I.e. should we continue to propagate 4xx / 5xx errors to the FE and leave the FE to decide how to handle them?

Comment (RyanKellyHMRC) [reply app/uk/gov/hmrc/plasticpackagingtaxregistration/connectors/SubscriptionsConnector.scala:59]: I say go with what you have. We can also change it later if we decide we need to see this stuff in the FE but I very much doubt we will.

File: app/uk/gov/hmrc/plasticpackagingtaxregistration/connectors/models/eis/subscriptionStatus/SubscriptionStatusResponse.scala [hunk 2]
```diff
@@ -17,18 +17,31 @@
...
+case class SubscriptionStatusResponse(status: Status, pptReference: Option[String] = None)
 
 object SubscriptionStatusResponse {
   implicit val format: OFormat[SubscriptionStatusResponse] = Json.format[SubscriptionStatusResponse]
+
+  def fromETMPResponse(etmpResponse: ETMPSubscriptionStatusResponse) = {
+
+    val status = etmpResponse.subscriptionStatus match {
+      case Some(NO_FORM_BUNDLE_FOUND) => NOT_SUBSCRIBED
+      case Some(SUCCESSFUL)           => SUBSCRIBED
```
Comment (RyanKellyHMRC) [inline app/uk/gov/hmrc/plasticpackagingtaxregistration/connectors/models/eis/subscriptionStatus/SubscriptionStatusResponse.scala:40]: I'm not sure that the translation from `SUCCESSFUL` to `SUBSCRIBED` is tested anywhere?

**Review by RyanKellyHMRC — COMMENTED** [review]
I think it looks good, definitely an improvement. Just a query about test coverage.

**Review by paulblakehmrc — APPROVED** [review]

**Review by RyanKellyHMRC — APPROVED** [review]

File: app/uk/gov/hmrc/plasticpackagingtaxregistration/connectors/models/eis/subscriptionStatus/SubscriptionStatusResponse.scala (hunk 2 above)
Comment (tmc-ee) [reply app/uk/gov/hmrc/plasticpackagingtaxregistration/connectors/models/eis/subscriptionStatus/SubscriptionStatusResponse.scala:40]: @RyanKellyHMRC This is now resolved here: https://github.com/hmrc/plastic-packaging-tax-registration/pull/57/files#diff-4eeaeffb89ffa2d4b42de4c6e695e33723f290beaca4221717006ea5600b68bdR33

**Review by tmc-ee — APPROVED** [review]

**Pull request merged**